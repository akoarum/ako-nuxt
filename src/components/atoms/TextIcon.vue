<template>
  <i :class="styledClass" class="textIcon"><slot /></i>
</template>

<script>
export default {
  props: {
    primary: { type: Boolean },
    secondary: { type: Boolean },
    caution: { type: Boolean }
  },
  computed: {
    styledClass() {
      if (this.primary) return '-primary'
      if (this.secondary) return '-secondary'
      if (this.caution) return '-caution'
      return ''
    }
  }
}
</script>

<style lang="scss" scoped>
.textIcon {
  display: inline-block;
  background: #eaeaea;
  line-height: 1;
  font-style: normal;

  @include media-min {
    padding: 5px 7px;
    font-size: 1.4rem;
  }

  @include media-max {
    padding: 3px 7px;
    font-size: 1.2rem;
  }
}

.-primary {
  background: $color-theme;
  color: #fff;
}

.-secondary {
  background: $color-secondary;
  color: #fff;
}

.-caution {
  background: $color-caution;
  color: #fff;
}
</style>
