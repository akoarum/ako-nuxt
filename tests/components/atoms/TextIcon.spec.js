import { shallowMount } from '@vue/test-utils'
import TextIcon from '~/components/atoms/TextIcon'

describe('TextIcon', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(TextIcon, {
      propsData: {}
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('computed', () => {
    it('styledClass: propsからクラスを生成して返す', () => {
      expect(TextIcon.computed.styledClass.call({ primary: true })).toBe('-primary')
      expect(TextIcon.computed.styledClass.call({ secondary: true })).toBe('-secondary')
      expect(TextIcon.computed.styledClass.call({ caution: true })).toBe('-caution')
      expect(TextIcon.computed.styledClass.call({})).toBe('')
    })
  })
})
