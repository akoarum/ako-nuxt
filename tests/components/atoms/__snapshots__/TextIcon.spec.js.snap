// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextIcon initialized snapshot 1`] = `
<i
  class="textIcon"
/>
`;
